decidim-api/lib/decidim/api/test/component_context.rb

```diff
@@ -42,3 +42,185 @@
+
+    context "when component is published" do
+      let!(:current_component) { create(component_factory, :published, participatory_space: participatory_process) }
+
+      context "when the user is admin" do
+        let!(:current_user) { create(:user, :admin, :confirmed, organization: current_organization) }
+
+        it "should be visible" do
+          expect(response["participatoryProcess"]["components"].first[lookout_key]).to eq(query_result)
+        end
+      end
+
+      context "when user is visitor" do
+        let!(:current_user) { nil }
+
+        it "should be visible" do
+          expect(response["participatoryProcess"]["components"].first[lookout_key]).to eq(query_result)
+        end
+      end
+
+      context "when user is normal user" do
+        let!(:current_user) { create(:user, :confirmed, organization: current_organization) }
+
+        it "should be visible" do
+          expect(response["participatoryProcess"]["components"].first[lookout_key]).to eq(query_result)
+        end
+      end
+    end
+
+    context "when component is not published" do
+      let!(:current_component) { create(component_factory, :unpublished, participatory_space: participatory_process) }
+
+      context "when the user is admin" do
+        let!(:current_user) { create(:user, :admin, :confirmed, organization: current_organization) }
+
+        it "should be visible" do
+          expect(response["participatoryProcess"]["components"].first[lookout_key]).to eq(query_result)
+        end
+      end
+
+      context "when user is visitor" do
+        let!(:current_user) { nil }
+
+        it "should not be visible" do
+          expect(response["participatoryProcess"]["components"].first).to be_nil
+        end
+      end
+
+      context "when user is normal user" do
+        let!(:current_user) { create(:user, :confirmed, organization: current_organization) }
+
+        it "should not be visible" do
+          expect(response["participatoryProcess"]["components"].first).to be_nil
+        end
+      end
+    end
+  end
+  context "when space is published but private" do
+    let!(:participatory_process) { create(process_space_factory, :published, :private, :with_steps, organization: current_organization) }
+
+    context "when component is published" do
+      let!(:current_component) { create(component_factory, :published, participatory_space: participatory_process) }
+
+      context "when the user is admin" do
+        let!(:current_user) { create(:user, :admin, :confirmed, organization: current_organization) }
+
+        it "should be visible" do
+          expect(response["participatoryProcess"]["components"].first[lookout_key]).to eq(query_result)
+        end
+      end
+
+      context "when user is visitor" do
+        let!(:current_user) { nil }
+
+        it "should not be visible" do
+          expect(response["participatoryProcess"]).to be_nil
+        end
+      end
+
+      context "when user is normal user" do
```

Comment:
We need another context for "when user is member" (should be visible)

Reply:
done in bd1c077a94